### PR TITLE
DatePicker

### PR DESCRIPTION
**Pull request by josephhajduk**
includes RangePicker, WeekPicker, MonthPicker

includes bs-moment in dependencies

usage

```
<Row>
  <Col span=6>
    <WeekPicker renderExtraFooter={()=>(ReasonReact.stringToElement("EXTRA W FOOTER"))}/>
  </Col>
  <Col span=6>  
    <MonthPicker renderExtraFooter={()=>(ReasonReact.stringToElement("EXTRA M FOOTER"))}/>
  </Col>
  <Col span=6>
    <DatePicker value=MomentRe.momentNow() showTime=DatePicker.Bool(true) format="YYYY-MM-DD HH:mm:ss" renderExtraFooter={()=>(ReasonReact.stringToElement("EXTRA FOOTER"))}/>
  </Col>
  <Col span=6>
    <RangePicker showTime=DatePicker.Bool(true) format="YYYY-MM-DD HH:mm:ss" renderExtraFooter={()=>(ReasonReact.stringToElement("EXTRA R FOOTER"))}/>
  </Col>
</Row>
```

### CODE REVIEW

File: src/Antd_TimePicker.re [hunk 1]
```diff
@@ -0,0 +1,8 @@
+type timePickerProps('a) = Js.t({. timePickerProps: int} as 'a);
```
Comment (josephhajduk) [inline src/Antd_TimePicker.re:1]: this is kinda hacky, could change it to just

```type timePickerProps('a) = Js.t('a);```

but then you lose some type safety as you could do

```<DatePicker showTime=TimePickerProps(someOtherSpecialCreationFunction(~foo="bar",())/>```

Comment (thangngoc89) [reply src/Antd_TimePicker.re:1]: type t is a convention in OCaml. It should be use instead. And I think this is not better than identity hack.

Comment (josephhajduk) [reply src/Antd_TimePicker.re:1]: how about 
```
module TimePickerProps {
  type t('a) = Js.t({. timePickerProps: int} as 'a);

  [@bs.obj]
  external make :
    (
      ~size: string=?,
      ~value: moment=?,
      ~defaultValue: moment=?, /* should be moment | moment[] */
      ~_open: Js.boolean=?,
      ~format: string=?,
      ~onChange: (moment, string) => unit=?,    
      ~onOpenChange: Js.boolean => unit=?,
      ~disabled: Js.boolean=?,
      ~placeholder: string=?,
      ~prefixCls: string=?,
      ~hideDisabledOptions: Js.boolean=?,
      ~disabledHours: unit => array(int)=?,
      ~disabledMinutes: int => array(int)=?,
      ~disabledSEconds: (int,int) => array(int)=?,
      ~getPopupContainer: Dom.element => Dom.htmlElement=?,
      ~addon: 'a=?,
      ~use12Hours: Js.boolean=?,
      ~focusOnOpen: Js.boolean=?,
      ~hourStep: Js.boolean=?,
      ~minuteStep: Js.boolean=?,
      ~secondStep: Js.boolean=?,
      ~allowEmpty: Js.boolean=?,
      ~clearText: Js.boolean=?,
      ~defaultOpenValue: moment=?,
      ~popupClassName: string=?,
      ~id: string=?,
      ~className: string=?,
      ~style: ReactDOMRe.Style.t=?,
      unit
    ) =>
    t(_) =
    "";
};```

Comment (thangngoc89) [reply src/Antd_TimePicker.re:1]: I don’t have an opinion on this actually. Both solutions are hacks so keeping the current type is fine with me. Anyway, is this type be reused elsewhere?

Comment (josephhajduk) [reply src/Antd_TimePicker.re:1]: Ya it is used in the showTime prop for DatePicker

**Comment by josephhajduk** [comment]
im going to close this and reopen when TimePicker is also ready as they sort of go hand in hand.

**Comment by thangngoc89** [comment]
YOu can keep this open and push new commit to this branch, i won’t merge until you said it’s ready. Finger crossed

**Comment by josephhajduk** [comment]
Is this usage desirable?

```<DatePicker showTime=DatePicker.TimePickerProps(TimePickerProps.make(~minuteStep=30,()))/>```

showTime can be a bool or TimePickerProps.t

should we consider splitting props like these up?  they seem to be a common pattern in ant-design (i.e. ColProps in Col seems to be used in Form)

perhape have a showTime:bool and a showTimeProps:TimePickerProps.t ?
could add some check to the make function to ensure both arn't Some(_)

then we would have
```<DatePicker showTimeProps=TimePickerProps.make(~minuteStep=30,())/>```

or 

```<DatePicker showTime=true/>```

**Comment by thangngoc89** [comment]
I have a dream of rewriting this whole thing in Reasonml to make unpresentable state impossible. Only keep stylesheets. Anyway, i prefer the first method. What if user defines both props by accident? 

I have typed ColProps for usage in Form. I’ll push it tonight

**Pull request closed without merging**